### ISSUE TITLE
Probability: rewrite as Probability

### ISSUE THREAD

**Issue by Upabjojr** [sympy/sympy#11136]
Allow to rewrite symbolic probability expressions as probability.

Should distinguish two cases:

 * continous RV ===> integral ... P(x) dx
 * discrete and finite RV ===> sum ... P(x) 

Possibly: give an option to use `DiracDelta` and integral for the discrete case.

**Comment by Smit-create** [comment]
Hi @Upabjojr,
Currently, the symbolic classes `Expectation`, `Variance`, and `Covariance` are supported with `.rewrite(Probability)` as:
https://github.com/sympy/sympy/blob/e926fc095fd6402f4d01dc4bf119e63e9fb56a76/sympy/stats/symbolic_probability.py#L222-L227

https://github.com/sympy/sympy/blob/e926fc095fd6402f4d01dc4bf119e63e9fb56a76/sympy/stats/symbolic_probability.py#L357-L360

https://github.com/sympy/sympy/blob/e926fc095fd6402f4d01dc4bf119e63e9fb56a76/sympy/stats/symbolic_probability.py#L493-L496

Also, in the symbolic `Probability` class, we can rewrite it as `Integral` or `Sum` as:
https://github.com/sympy/sympy/blob/e926fc095fd6402f4d01dc4bf119e63e9fb56a76/sympy/stats/symbolic_probability.py#L61-L64

Does anything more needs to be added?

**Comment by Upabjojr** [comment]
Do we have "rewrite as variance"?

**Comment by Smit-create** [comment]
> Do we have "rewrite as variance"?

I didn't get your point as How we could write Expectation in terms of Variance as Var(x) = E(x**2) - E(x)**2? Means, how can we write all others in terms of variance. Can you please provide some references?

**Comment by Upabjojr** [comment]
Transforming variance to expectation is easy... just replace with the formula `Var(x) = E(x**2) - E(x)**2`.

But `E(x**2) - E(x)**2` could also be transformed into `Var(x)`.

This second transformation is a bit more complicated to achieve... but maybe one can create an algorithm.

There are also more complicated cases such as `Var(x + y) = E(x**2) + E(y**2) - E(x)**2 - E(y)**2 + 2*E(x*y) - 2*E(x)*E(y)`.

If you have only part of that expression, e.g. `E(x**2) + E(y**2) - E(x + y)**2`, what should you do?

[OUT OF TOPIC] Do we have a proper way to fully expand the `Expectation`? I mean:
```python
In [13]: Variance(X+Y).rewrite(Expectation)
Out[13]:
                    2
- Expectation(X + Y)  + Expectation((X + Y)**2)

In [14]: _.expand()
Out[14]:
                    2
- Expectation(X + Y)  + Expectation(X**2 + 2*X*Y + Y**2)
```

**Comment by Smit-create** [comment]
> Do we have a proper way to fully expand the `Expectation`? I mean:

Yes, we have, but `expand` of `Expr` doesn't pass on the call to `expand` of `Expectation` when the `_.expand()` is called on `Add` object.

On the currrent master:
```py
>>> Expectation((X + Y)**2).expand()
Expectation((X + Y)**2)
```

But this can be fixed to produce:
```py
>>> Expectation((X + Y)**2).expand()
2*Expectation(X*Y) + Expectation(X**2) + Expectation(Y**2)
```

**Comment by Upabjojr** [comment]
Yes, needs fixing.
